MoltenVK/MoltenVK/GPUObjects/MVKImage.h

```diff
@@ -195,7 +195,12 @@ class MVKImage : public MVKResource {
 	/** Returns the Metal CPU cache mode used by this image. */
 	inline MTLCPUCacheMode getMTLCPUCacheMode() { return _deviceMemory->getMTLCPUCacheMode(); }
 
-	
+	/** 
+	 * Returns whether the memory is automatically coherent between device and host. 
+	 * On macOS, this always returns false because textures cannot use Shared storage mode.
+	 */
+	bool isMemoryHostCoherent();
```

Comment:
Hey wait, doesn't MVKResource already have this method? I find it curious that adding this method here, which hides the method in the base, and doing nothing else, fixes your problem.

Reply:
@cdavis This is a slight variation in that calling `MVKImage:: getMTLStorageMode()` deals with swapping out `MTLStorageModeShared` on macOS.

Reply:
Ah, I see. Thanks for clearing that up.